File: Majority_Element.java
import java.util.Arrays;

public class Majority_Element {
    public static void main(String[] args) {
        int[] arr = { 2, 3, 3, 2, 3, 2, 2, 2 };
        int res = majorityEle(arr);
        System.out.println(res);

    }

    public static int majorityEle(int[] nums) {
        int count = 0;
        int max_ele = 0;
        for (int i = 0; i < nums.length; i = i + 1) {
            if (count == 0) {
                max_ele = nums[i];
            }
            if (nums[i] == max_ele) {
                count++;
            } else {
                count--;
            }

        }
        return max_ele;

    }

}

File: MajorityElement/majorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
    int index = 0;
    int count = 1;
    for (int i = 1; i <nums.size(); i++) {
        if (nums[i] == nums[index]) count++;
        else count--;
        if (count == 0) {
            index = i;
            count = 1;
        }
    }
    if (count > 0) {
        int c = 0;
        int ele = nums[index];
        for (int i = 0; i <nums.size(); i++) {
            if (nums[i] == ele) c++;
        }
        if (c > nums.size() / 2) return(ele);
    }
    return(-1);
    }
};